### PR TITLE
Losses.md docs

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

File: documentation/assets/Losses.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    ...
+    
+    loss: yolox_loss
+    
+    criterion_params:
+      strides: [8, 16, 32]  # output strides of all yolo outputs
+      num_classes: 80
+  ```
+  
+   `criterion_params` dictionary will be unpacked to the underlying `yolox_loss` class constructor.
```
Comment (ofrimasad) [inline documentation/assets/Losses.md:37]: Dont start a sentence like this. 
start here with: "Note that two parameters define the loss function... loss: defines the type of the loss and..."

File: documentation/assets/Losses.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  
+        train_params = {
+            ...
+            "loss": "cross_entropy",
+            "criterion_params": {}
+            ...
+        }
+        trainer.train(model=model, training_params=train_params, train_loader=train_dataloader, valid_loader=valid_dataloader)
+   ```
+   Note that object names in SG are not case-sensitive nor symbol-sensitive, so `"CrossEntropy` could have been passed as well.
```
Comment (ofrimasad) [inline documentation/assets/Losses.md:57]: this is great addition

File: documentation/assets/Losses.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+```python
+import torch.nn
+
+MyLoss(torch.nn.Module):
+...
+forward(preds, target):
+...
+
+```
+And as the argument names suggest- the first argument is the model's output, and target is the label/ground truth (argument naming is arbitrary and does not need to be specifically 'preds' or 'target').
```
Comment (ofrimasad) [inline documentation/assets/Losses.md:79]: "suggest-"  -> "suggest,"

Comment (ofrimasad) [reply documentation/assets/Losses.md:79]: what about the expected output format?
please add something about `component_names` and returning the loss components

**Review by ofrimasad — COMMENTED** [review]
looks very nice. 
my comment inline

File: documentation/assets/Losses.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
+# Losses in SG
+
+SuperGradients provides multiple Loss function implementations for various tasks:
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/assets/Losses.md]: added https://github.com/Deci-AI/super-gradients/pull/658/commits/e6cc16d897d78a095f5fce559fa70957a8ddd783

File: documentation/assets/Losses.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    stdc_loss
+    bce_dice_loss
+    kd_loss
+    dice_ce_edge_loss
+
+All of the above, are just string aliases for the underlying torch.nn.Module classes, implementing the specified loss functions.
+
+##Basic Usage of Implemented Loss Functions in SG:
+
+When using configuration files, for example training using train_from_recipe (or similar, when the underlying train method that is being called is Trainer.train_from_config(...)):
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/assets/Losses.md]: changed the order in https://github.com/Deci-AI/super-gradients/pull/658/commits/e6cc16d897d78a095f5fce559fa70957a8ddd783

File: documentation/assets/Losses.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    yolox_fast_loss
+    ssd_loss
+    stdc_loss
+    bce_dice_loss
+    kd_loss
+    dice_ce_edge_loss
+
+All of the above, are just string aliases for the underlying torch.nn.Module classes, implementing the specified loss functions.
+
+##Basic Usage of Implemented Loss Functions in SG:
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/assets/Losses.md]: fixed https://github.com/Deci-AI/super-gradients/pull/658/commits/e6cc16d897d78a095f5fce559fa70957a8ddd783

File: documentation/assets/Losses.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    criterion_params:
+      strides: [8, 16, 32]  # output strides of all yolo outputs
+      num_classes: 80
+  ```
+  
+   `criterion_params` dictionary will be unpacked to the underlying `yolox_loss` class constructor.
+
+
+Another usage case, is when using a direct Trainer.train(...) call:
+   - In your `my_training_script.py`
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/assets/Losses.md]: https://github.com/Deci-AI/super-gradients/pull/658/commits/e6cc16d897d78a095f5fce559fa70957a8ddd783 done

File: documentation/assets/Losses.md (hunk 1 above)
Comment (shaydeci) [reply documentation/assets/Losses.md]: changed in https://github.com/Deci-AI/super-gradients/pull/658/commits/2d799e1723e68f36e89b1117bc8f5d8fd237f0d0

File: documentation/assets/Losses.md (hunk 3 above)
Comment (shaydeci) [reply documentation/assets/Losses.md]: fixed and re-ordered in 2 last commits.

**Review by ofrimasad — APPROVED** [review]
LGTM
